frontend/src/lib/createIdFromObjectFields.ts

```diff
@@ -0,0 +1,11 @@
+import { idDivider } from 'constants/queryBuilder';
+
+export const createIdFromObjectFields = <T, K extends keyof T>(
+	obj: T,
+	keys?: K[],
+): string => {
+	const currentObj = obj as Record<string, unknown>;
+	const currentKeys = keys ?? (Object.keys(currentObj) as K[]);
+
+	return currentKeys.map((key) => obj[key]).join(idDivider);
+};
```

Comment:
This `Object.keys` will be problematic in dashboards since dashboard JSON also stores the `id`. This will lead to incorrect ref. Try to edit in dashboards and see it fail. I suggest being explicit about the keys. For example, only use `key`, `dataType`, `type`, `IsColumn` and remove `Object.keys`
<img width="1408" alt="Screenshot 2023-06-02 at 7 22 55 AM" src="https://github.com/SigNoz/signoz/assets/22846633/f06f2bfd-b062-4415-9726-0ce69e443c7b">

Reply:
Loom https://www.loom.com/share/af0cf1c3cffa46c38cba266c86573911